src/Lean/Elab/Tactic/BuiltinTactic.lean

```diff
@@ -364,7 +364,9 @@ partial def evalChoiceAux (tactics : Array Syntax) (i : Nat) : TacticM Unit :=
 where
   introStep (ref : Option Syntax) (n : Name) (typeStx? : Option Syntax := none) : TacticM Unit := do
     let fvarId ← liftMetaTacticAux fun mvarId => do
-      let (fvarId, mvarId) ← mvarId.intro n
+      let (fvarId, mvarId) ← match ref with
+        | none     =>                mvarId.intro n
+        | some ref => withRef ref <| mvarId.intro n
```

Comment:
I wouldn’t mind a `Lean.withRef?` variant of `withRef` that makes this code here prettier and would allow
```suggestion
      let (fvarId, mvarId) ← withRef? ref <| mvarId.intro n
```
and maybe rename the `ref` variable to `ref?` while you are it (common idiom for Option-typed variables)

Reply:
Excellent suggestion. Done.